asmcomp/amd64/emit.mlp

```diff
@@ -435,17 +456,18 @@ let emit_float_test cmp i lbl =
 
 let output_epilogue f =
   if frame_required() then begin
-    let n = frame_size() - 8 - (if fp then 8 else 0) in
-    if n <> 0
-    then begin
-      I.add (int n) rsp;
-      cfi_adjust_cfa_offset (-n);
+    let frame_size = frame_size () in
+    let n = frame_size - 8 - (if fp then 8 else 0) in
+    add_to_rsp n;
+    if fp then begin
+      pop rbp
     end;
-    if fp then I.pop rbp;
     f ();
     (* reset CFA back cause function body may continue *)
-    if n <> 0
-    then cfi_adjust_cfa_offset n
+    cfi_adjust_cfa_offset ~bytes:n;
+    if fp then begin
+      cfi_adjust_cfa_offset ~bytes:8
+    end
```

Comment:
I can see that this "undoes" the stack manipulations before the call to `f ()`, but just for my understanding, could you say a few words as to why this code is necessary, given that `f ()` always jumps/returns away from the current function?